api/lib/domain/usecases/attach-organizations-from-existing-target-profile.js

```diff
@@ -0,0 +1,23 @@
+const _ = require('lodash');
+const { NoOrganizationToAttach } = require('../errors');
+
+module.exports = async function attachOrganizationsFromExistingTargetProfile({
+  targetProfileId,
+  existingTargetProfileId,
+  organizationRepository,
+  targetProfileRepository,
+}) {
+
+  const targetProfile = await targetProfileRepository.get(targetProfileId);
+
+  // check if existingTargetProfileId exists, if not throws
+  await targetProfileRepository.get(existingTargetProfileId);
+
+  const organizationIds = await organizationRepository.findIdsByTargetProfileId(existingTargetProfileId);
+  if (_.isEmpty(organizationIds)) {
+    throw new NoOrganizationToAttach(`Le profil cible ${existingTargetProfileId} n'a aucune organisation rattachée.`);
+  }
+
+  targetProfile.addOrganizations(organizationIds);
+  await targetProfileRepository.attachOrganizations(targetProfile);
```

Comment:
**suggestion (non-blocking) :** C'est un peu dommage ce use case, on a un get qui est un check et on récupère les organisation après alors qu'il ça pourrait être mieux de le faire avec `await targetProfileRepository.get(existingTargetProfileId);` 

 J'imagine un truc comme ça :
```js
  const targetProfile = await targetProfileRepository.get(targetProfileId);
  const targetProfileToCopy = await targetProfileRepository.get(existingTargetProfileId);

  targetProfile.copyAttachments(targetProfileToCopy)

  await targetProfileRepository.attachOrganizations(targetProfile);
```

Reply:
out of date, j'ai réduit à ça :
```
  const organizationIds = await targetProfileRepository.findOrganizationIds(existingTargetProfileId);
  if (_.isEmpty(organizationIds)) {
    throw new NoOrganizationToAttach(`Le profil cible ${existingTargetProfileId} n'a aucune organisation rattachée.`);
  }

  await targetProfileRepository.attachOrganizationIds({ targetProfileId, organizationIds });
  ```